packages/fxa-shared/test/metrics/amplitude.js

```diff
@@ -644,6 +685,82 @@ describe('metrics/amplitude:', () => {
       }
     });
 
+    it('errors - if event_type is `fxa_pay_setup - *` and country_iso is more than 2 characters', () => {
+      const event = {
+        ...minimumEvent,
+        event_type: 'fxa_pay_setup - view',
+        language: 'en',
+        country_iso: 'USA',
+      };
+      try {
+        amplitude.validate(event);
+        assert.fail('Validate is expected to fail');
+      } catch (err) {
+        assert.isTrue(err instanceof Error);
+        assert.equal(
+          err.message,
+          `Invalid data: event/country_iso must match pattern "^[A-Z]{2}$", event must match "then" schema`
+        );
+      }
+    });
+
+    it('errors - if event_type is `fxa_pay_subscription_change - *` and country_iso is more than 2 characters', () => {
+      const event = {
+        ...minimumEvent,
+        event_type: 'fxa_pay_subscription_change - view',
+        language: 'en',
+        country_iso: 'USA',
+      };
+      try {
+        amplitude.validate(event);
+        assert.fail('Validate is expected to fail');
+      } catch (err) {
+        assert.isTrue(err instanceof Error);
+        assert.equal(
+          err.message,
+          `Invalid data: event/country_iso must match pattern "^[A-Z]{2}$", event must match "then" schema`
+        );
+      }
+    });
+
+    it('errors - if event_type is `fxa_pay_setup - *` and country_iso less than 2 characters', () => {
+      const event = {
+        ...minimumEvent,
+        event_type: 'fxa_pay_setup - view',
+        language: 'en',
+        country_iso: 'U',
+      };
+      try {
+        amplitude.validate(event);
+        assert.fail('Validate is expected to fail');
+      } catch (err) {
+        assert.isTrue(err instanceof Error);
+        assert.equal(
+          err.message,
+          `Invalid data: event/country_iso must match pattern "^[A-Z]{2}$", event must match "then" schema`
+        );
+      }
+    });
+
+    it('errors - if event_type is `fxa_pay_subscription_change - *` and country_iso is less than 2 characters', () => {
+      const event = {
+        ...minimumEvent,
+        event_type: 'fxa_pay_subscription_change - view',
+        language: 'en',
+        country_iso: 'U',
+      };
+      try {
+        amplitude.validate(event);
+        assert.fail('Validate is expected to fail');
+      } catch (err) {
+        assert.isTrue(err instanceof Error);
+        assert.equal(
+          err.message,
+          `Invalid data: event/country_iso must match pattern "^[A-Z]{2}$", event must match "then" schema`
+        );
+      }
+    });
+
```

Comment:
Not going to make you change anything here, but something to keep in mind if/when we add more global required/optional properties is to make a list of event names, a list of required properties and a list of optional properties and iterate through that list and perform the same set of tests for all required properties and the same set of tests for all optional properties. That way we ensure we cover every permutation.